src/main/java/org/javarosa/xpath/expr/XPathPathExpr.java

```diff
@@ -84,6 +86,7 @@ public TreeReference getReference() throws XPathUnsupportedException {
                 break;
             case XPathPathExpr.INIT_CONTEXT_RELATIVE:
                 ref.setRefLevel(0);
+                ref.setContextType(TreeReference.CONTEXT_INHERITED);
```

Comment:
this fixes a bug where `XPathPathExpr.getReference()` was not properly assigning the context type here. The reason this wasn't an issue for anything else is that there was previously no logic in the codebase which hinged on checking for the value `CONTEXT_INHERITED` (this PR changes that)